### ISSUE TITLE
Support multimode displays

### ISSUE THREAD

**Issue by caco3** [jomjol/AI-on-the-edge-device#1936]
### The Feature

There are energy meters with digital displays which alternate to display different informations.
Eg:
 - Value of low tariff
 - Value of high tariff
 - Time
 - ...

They usually automatically iterate through those modes based on a time interval.
They usually have a spot on the display which shows the currently showed mode.

There where now several requests for such thing, so we should consider looking into it.

Sources:
- https://github.com/jomjol/AI-on-the-edge-device/discussions/1920
-

**Comment by jonas240** [comment]
Another example would be something like this: https://qundis.de/wp-content/uploads/2017/12/Datenblatt_Q-water-5.5_Messkapsel_V220.pdf
The “important” part, which shows the cumulative consumption, looks like this.
![grafik](https://user-images.githubusercontent.com/68380809/217052441-1a9445c9-b9fe-4cd6-bdbe-c9cbe09339c6.png)
However, there is also a view, which shows last year's total consumption, which looks rather similar in terms of “mode indication”:
![grafik](https://user-images.githubusercontent.com/68380809/217053361-f08781f5-c955-421c-b833-413963fc9d80.png)
Would the detection of the “cumulative consumption view” still work in that case?

**Comment by caco3** [comment]
Is the only difference the bottom right arrow?
It might be quite tricky to detect such thing reliably.
And it for sure would have to be specifically trained/configured.

**Comment by jonas240** [comment]
Yes. Another indicator would be that last year's total consumption is always less. But I am not well enough informed about this project to know if that can be considered with how everything is processed.

**Comment by caco3** [comment]
Technically it is posssible, but I think it is quite dome effort.

some solutions which might work for now:
- disable negativeRateCheck and maxRateCheck and postprocess/split on receiver side. it should not be too hardveg. with homeassistant based on the numbers if they are far enough apart.
- or place 2 NUMBERS above each other, both with its matching preValue. always one then will have an error but the other one will transmit its value.

**Comment by robin13** [comment]
I'd like to add another related usecase here. In the above situations, there are multiple values being displayed in sequence, but only one of those values is important.
In my situation I have an Iskra MT631 (https://www.iskraemeco.com/app/uploads/2019/09/MT631-632.pdf) electricity meter which alternates between showing the produced (photovoltaik surplus) energy and the used energy.
It would be very useful to capture **both** of these numbers.
The display looks like this, and only the "OBIS-Identification" (1) changes from `1.8.0` to `2.8.0`
![2023-02-23_07-12](https://user-images.githubusercontent.com/174291/220832771-2dde9589-6d0d-481e-878d-0568e29ae45b.png)

I could imagine the workflow where the ROI configuration has an additional class "meter key" where the critical area(s) can be highlighted, and examples can be made by which the image types can be differentiated.
- Capture image and do alignment
- Determine "meter key" based on best match of the corresponding ROIs
- Apply the corresponding digitisation
- Send the data of the captured digitised value together with the "meter key"

Let me know if you need any further information, sample pictures etc. to help with this.

**Comment by caco3** [comment]
Another request: https://github.com/jomjol/AI-on-the-edge-device/discussions/2183

**Comment by TravisRoy** [comment]
I am also having this issue with my new water meter. It works WAY better than my old meter, but it does not like it when the display changes. 

<img width="1017" alt="Screenshot 2024-07-31 at 6 18 09 PM" src="https://github.com/user-attachments/assets/70839976-bdb1-45c6-b69b-6bab8eb54fdb">

**Comment by TravisRoy** [comment]
Meter documentation here - https://www.instrumart.com/assets/HR-E-manual.pdf

**Comment by caco3** [comment]
Which model are you using? Did you give https://jomjol.github.io/AI-on-the-edge-device-docs/Choosing-the-Model/#dig-class11 a try? Looking on the documentation, this actually should solve your issue by detecting blank displays as NaN. 

Just to make you aware, but AFAIK there currently is no developer working on this feature request.
Also I think it is quite tricky. The neuronal network works in the way that it searches the best fitting image and uses the corresponding stored value. It would somehow needed to be extended with a threshold which makes sure only "clear" (how ever this would be qualified) detections are used...

@TravisRoy The contrast is very poor, this also will be an issue!

**Comment by SybexX** [comment]
https://github.com/jomjol/AI-on-the-edge-device/issues/2989

**Comment by caco3** [comment]
There are more and more request for such feature, eg. https://github.com/jomjol/AI-on-the-edge-device/discussions/3141

As of now, nobody is working on such feature, it would quite some work.
But if somebody would want to do it, we would need some raw images of a meter in the various modes so we could test it.
So if somebody is able to provide good test images, that would be a start.
See https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-rawimageslocation

@leneffets, @sweti3, @NTohan FYI

**Comment by lukeyo** [comment]
I think I have solved the problem, at least for smart meters that display 1.8.0, 1.8.1, and 1.8.2. 

I simply always read this number, i.e., it results in something like 1803757284.

There, I can then subtract the number from 1800000000 in Home Assistant and get 3757284, which is the actual reading. 

The numbers are also relatively far apart, so they don't accidentally overwrite each other. I have also created three number sequences that always read the same thing. 

I have set the round interval to 3.28, so I always oscillate between two values and then calculate the third.

I have been using this for about two years now, and it works consistently, except when the ESP falls down on the ground.

**Comment by alexschultze** [comment]
Another digital use case and example image here. This time for Honeywell Digital Water meters.

Despite the meter beeing digital and even radio (wmBus) some utility companies apparently dont like to hand out or change the keys (since they use the same key for all their water meters). 

<img width="664" height="1036" alt="Image" src="https://github.com/user-attachments/assets/7532215c-48e8-4e40-9a68-0b6b54e3bbe1" />
